src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/handlers/VideoBookingCancelledEventHandler.kt

```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.handlers
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.PrisonAppointmentRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoBookingRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.VideoBookingCancelledEvent
+
+@Component
+class VideoBookingCancelledEventHandler(
+  private val videoBookingRepository: VideoBookingRepository,
+  private val prisonAppointmentRepository: PrisonAppointmentRepository,
+) : DomainEventHandler<VideoBookingCancelledEvent> {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  override fun handle(event: VideoBookingCancelledEvent) {
+    videoBookingRepository.findById(event.additionalInformation.videoBookingId).ifPresentOrElse(
+      { booking ->
+        prisonAppointmentRepository.findByVideoBooking(booking).forEach {
+          // TODO - raise delete event or just call the manage external appointments service directly here?
```

Comment:
We have a choice here to spawn off separate events to deal with removal of each appointment from external services.